lib/postprocessing/image.js

```diff
@@ -12,56 +12,55 @@
 
 'use strict';
 
-const gm = require("gm").subClass({ imageMagick: true });
-const { execSync } = require("child_process");
-const fs = require("fs-extra");
-const path = require("path");
-
-// default operations to apply to an image during image processing
-const DEFAULT_OPERATIONS = {
-    "basicProcessing": true,
-    "watermarking": false,
-    "removeMetadata": false
-};
-
-// Returns a temporary file name with the specified extension
-function tempFileName(fmt) {
-    const date = new Date();
-    const milli = date.getTime();
-    const tmpFile = `sdk-postprocess-${milli}.${fmt}`;
-    return tmpFile;
-}
+const gm = require("./gm-promisify");
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+const fs = require('fs').promises;
 
-// Gets file size
-function fileSize(file) {
-    const stats = fs.statSync(file);
-    return stats.size;
+async function logImageMagickVersion() {
```

Comment:
Do we need that in prod?

Reply:
It think it's very useful, because as we have seen, the imagemagick version is impossible to guess from the outside of an image (based on the OS) and we will always have some slight different images in prod, some workers might have a newer base version already, some not... Hence once we have a bug report about a particular rendition, we would know exactly the IM version right from the logs without having to do further forensics...

I am not sure about the cost of that `identify --version` call on every request. If we do that at startup of the container (i.e. upon loading of the module) we would not see it in the activation logs.

Reply:
`the imagemagick version is impossible to guess from the outside of an image (based on the OS)` it should be in the metadata imagemagick writes (as I recall with don't remove them).

Reply:
ok. do we know if that is added to all output formats or just jpeg?

Reply:
Regarding the performance: on my Mac with imagemagick 7.0.10 printing the version takes ~20 msec +/- 10. That looks not too bad.

```
> time identify --version
Version: ImageMagick 7.0.10-28 Q16 x86_64 2020-08-16 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(3.1) 
Delegates (built-in): bzlib freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps tiff webp xml zlib

real	0m0.021s
user	0m0.006s
sys	0m0.011s
```

One thing we could do is print this after the renditions have been uploaded so it is outside the "hot" code path, but still in the activation logs.

Reply:
I checked a png with cli exiftool and imagemagick's identify, it doesn't write into PNG metadata that it has been processed by imagemagick.